src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/ViewCodeCommandHandler.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using Microsoft.VisualStudio.Commanding;
+using Microsoft.VisualStudio.LanguageServerClient.Razor.VS.LSClientRazor;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    [Name(nameof(ViewCodeCommandHandler))]
+    [Export(typeof(ICommandHandler))]
+    [ContentType(RazorLSPConstants.RazorLSPContentTypeName)]
+    internal sealed class ViewCodeCommandHandler : ICommandHandler<ViewCodeCommandArgs>
+    {
+        private const string CSharpFileExtension = ".cs";
+
+        // Because query status happens all the time we want to cache the File.Exists checks for a reasonable amount of time
+        private const int CacheTimeoutMilliseconds = 10000;
+        private static readonly Stopwatch s_fileExistsStopwatch = Stopwatch.StartNew();
+        private static readonly Dictionary<string, (bool exists, long addedMs)> s_fileExistsCache = new();
```

Comment:
This is not  our normal naming convention, but it does kind of grow on me.

Reply:
Right? I like it too and would be down to make the change acrdoss the codebase. Way tooo easy to interact with a static and not know otherwise

Reply:
What is your normal naming convention? This is standard in the dotnet org AFAIK, but whoever puts the .editorconfig rule in first wins, I guess 😛

Reply:
In the past if it was private static we'd camel case, otherwise we'd pascal case (public static)